public/app/plugins/datasource/azuremonitor/dataquery.cue

```diff
@@ -0,0 +1,214 @@
+
+						#AzureMetricQuery: {
+							// Array of resource URIs to be queried.
+							resources?: [...#AzureMonitorResource]
+							// metricNamespace is used as the resource type (or resource namespace).
+							// It's usually equal to the target metric namespace. e.g. microsoft.storage/storageaccounts
+							// Kept the name of the variable as metricNamespace to avoid backward incompatibility issues.
+							metricNamespace?: string
+							// Used as the value for the metricNamespace property when it's different from the resource namespace.
+							customNamespace?: string
+							// The metric to query data for within the specified metricNamespace. e.g. UsedCapacity
+							metricName?: string
+							// The Azure region containing the resource(s).
+							region?: string
+							// The granularity of data points to be queried. Defaults to auto.
+							timeGrain?: string
+							// The aggregation to be used within the query. Defaults to the primaryAggregationType defined by the metric.
+							aggregation?: string
+							// Filters to reduce the set of data returned. Dimensions that can be filtered on are defined by the metric.
+							dimensionFilters?: [...#AzureMetricDimension]
+							// Maximum number of records to return. Defaults to 10.
+							top?: string
+							// Time grains that are supported by the metric.
+							allowedTimeGrainsMs?: [...int64]
+
+							// Aliases can be set to modify the legend labels. e.g. {{ resourceGroup }}. See docs for more detail.
+							alias?: string
+
+							// @deprecated
+							timeGrainUnit?: string
+
+							// @deprecated This property was migrated to dimensionFilters and should only be accessed in the migration 
+							dimension?: string
+
+							// @deprecated This property was migrated to dimensionFilters and should only be accessed in the migration 
+							dimensionFilter?: string
+
+							// @deprecated Use metricNamespace instead
+							metricDefinition?: string
+
+							// @deprecated Use resourceGroup, resourceName and metricNamespace instead 
+							resourceUri?: string
+
+							// @deprecated Use resources instead 
+							resourceGroup?: string
+							// @deprecated Use resources instead 
+							resourceName?: string
+						} @cuetsy(kind="interface")
+
+						// Azure Monitor Logs sub-query properties
+						#AzureLogsQuery: {
+							// KQL query to be executed.
+							query?: string
+							// Specifies the format results should be returned as.
+							resultFormat?: #ResultFormat
+							// Array of resource URIs to be queried.
+							resources?: [...string]
+							// Workspace ID. This was removed in Grafana 8, but remains for backwards compat
+							workspace?: string
+
+							// @deprecated Use resources instead 
+							resource?: string
+						} @cuetsy(kind="interface")
+
+						#ResultFormat: "table" | "time_series" @cuetsy(kind="enum")
```

Comment:
```suggestion
						#ResultFormat: "table" | "time_series" @cuetsy(kind="enum",memberNames="Table|TimeSeries")
```
Looks like we need to provide the member names as the ones with underscores (also dashes) aren't properly formatted.